src/init/scaffold.re

```diff
@@ -1000,13 +899,114 @@ module V2 = {
                </Box>
-             : <Spinner label="Building project with Dune" />}
+             : <Spinner
+                 label="Installing OCaml dev dependencies, this may take a few minutes"
+               />}
         </Box>;
       };
     };
   };
+};
+
+module Dune_install = {
+  open Ui;
+  [@react.component]
+  let make = (~state, ~onComplete, ~onError) => {
+    let (copy_complete, set_copy_complete) = React.useState(() => false);
+
+    let is_active =
+      state.step == Dune_install
+      && state.configuration.initialize_ocaml_toolchain;
+    let is_visible =
+      state.configuration.initialize_ocaml_toolchain
+      && step_to_int(state.step) >= step_to_int(Dune_install);
+
+    React.useEffect1(
+      () => {
+        if (is_active) {
+          state.configuration.directory
+          |> Engine.dune_install
+          |> Promise_result.perform(result =>
+               switch (result) {
+               | Ok(_) =>
+                 set_copy_complete(_ => true);
+                 onComplete();
+               | Error(err) => onError(err)
+               }
+             );
+        };
+
+        None;
+      },
+      [|is_active|],
+    );
+
+    if (!is_visible) {
+      React.null;
+    } else {
+      <Box flexDirection=`column gap=1>
+        {copy_complete
+           ? <Box flexDirection=`row gap=1>
+               <Badge color=`green> {React.string("Complete")} </Badge>
+               <Text> {React.string("Generating project opam file")} </Text>
+             </Box>
+           : <Spinner label="Generating project opam file" />}
+      </Box>;
+    };
+  };
+};
+
+module Dune_build = {
+  open Ui;
+  [@react.component]
+  let make = (~state, ~onComplete, ~onError) => {
+    let (copy_complete, set_copy_complete) = React.useState(() => false);
+
+    let is_active =
+      state.step == Dune_build
+      && state.configuration.initialize_ocaml_toolchain;
+    let is_visible =
+      state.configuration.initialize_ocaml_toolchain
+      && step_to_int(state.step) >= step_to_int(Dune_build);
+
+    React.useEffect1(
+      () => {
+        if (is_active) {
+          state.configuration.directory
+          |> Engine.dune_build
+          |> Promise_result.perform(result =>
+               switch (result) {
+               | Ok(_) =>
+                 set_copy_complete(_ => true);
+                 onComplete();
+               | Error(err) => onError(err)
+               }
+             );
+        };
+
+        None;
+      },
+      [|is_active|],
+    );
+
+    if (!is_visible) {
+      React.null;
+    } else {
+      <Box flexDirection=`column gap=1>
+        {copy_complete
+           ? <Box flexDirection=`row gap=1>
+               <Badge color=`green> {React.string("Complete")} </Badge>
+               <Text> {React.string("Building project with Dune")} </Text>
+             </Box>
+           : <Spinner label="Building project with Dune" />}
+      </Box>;
+    };
+  };
+};
 
   module Scaffold = {
```

Comment:
We can also remove this module/wrapper. That'll fix in `component.re` accessing it via `module Scaffold = Scaffold.Scaffold` and will instead be just  `module Scaffold = Scaffold`

Reply:
sure thing, will do and push in a bit

Reply:
should be good now :smile: